graphstorm-processing/scripts/run_distributed_processing.py

```diff
@@ -0,0 +1,265 @@
+
+    s3_train_config = (
+        f"{s3_input_prefix}{args.config_filename}"
+        if s3_input_prefix.endswith("/")
+        else f"{s3_input_prefix}/{args.config_filename}"
+    )
+    s3_input_bucket = s3_input_prefix.split("/")[2]
+    s3_input_key = s3_input_prefix.split("/", 3)[3]
+
+    if not s3_output_prefix:
+        s3_bucket = s3_input_prefix.split("/")[2]
+        s3_output_prefix = (
+            f"s3://{s3_bucket}/graphstorm-distributed-graph-processing/output/{processing_job_name}"
+        )
+
+    s3_output_bucket = s3_output_prefix.split("/")[2]
+    print(
+        "Creating SageMaker Processing job for train"
+        f" config in {s3_train_config} and outputs to {s3_output_prefix}"
+    )
+
+    # Create SageMaker session with high retry count to avoid thresholding errors
+    args.region = (
+        script_utils.get_bucket_region(s3_output_bucket) if not args.region else args.region
+    )
+    sm_boto, s3_boto = [
+        boto3.client(
+            service,
+            config=Config(
+                region_name=args.region,
+                connect_timeout=5,
+                read_timeout=60,
+                retries={"max_attempts": 20},
+            ),
+            region_name=args.region,
+        )
+        for service in ["sagemaker", "s3"]
+    ]
+    sagemaker_session = sagemaker.Session(
+        boto_session=boto3.Session(region_name=args.region), sagemaker_client=sm_boto
+    )
+
+    processor_kwargs = script_utils.parse_processor_kwargs(args.sm_processor_parameters)
+
+    byte_size_on_s3 = script_utils.determine_byte_size_on_s3(
+        s3_input_bucket, s3_input_key, s3_boto
+    ) // (1024 * 1024 * 1024)
+    input_total_size_in_gb = max(1, byte_size_on_s3 // (1024 * 1024 * 1024))
+    # Heuristic: total storage of 6+ times the total input size should be
+    # sufficient for Spark (assuming CSV input)
+    # Using anything less than 2*input_size/instance_count led to failures for large datasets
+    max_allowed_volume_size = script_utils.get_max_volume_size_for_processing(args.region)
+    if "volume_size_in_gb" not in processor_kwargs:
+        desired_volume_size = 6 * (input_total_size_in_gb // instance_count)
+    else:
+        desired_volume_size = processor_kwargs["volume_size_in_gb"]
+        processor_kwargs.pop("volume_size_in_gb")
+
+    if desired_volume_size > max_allowed_volume_size:
+        logging.warning(
+            "Desired volume size (%d GB) is larger than max required, assigning the max: %d GB",
+            desired_volume_size,
+            max_allowed_volume_size,
+        )
+    required_gb_per_instance = max(min(desired_volume_size, max_allowed_volume_size), 30)
+    logging.info(
+        "Total data size <= %d GB, "
+        "assigning %d GB storage "
+        "per instance (total storage: %d GB).",
+        input_total_size_in_gb,
+        required_gb_per_instance,
+        instance_count * required_gb_per_instance,
+    )
+
+    pyspark_processor = PySparkProcessor(
```

Comment:
Can you also PR PySparkProcessor?

Reply:
This is the internal SageMaker class to create PySpark jobs: https://sagemaker.readthedocs.io/en/stable/amazon_sagemaker_processing.html#pysparkprocessor